Gems/ROS2/Code/Source/Lidar/ROS2Lidar2DSensorComponent.cpp

```diff
@@ -172,28 +174,25 @@ namespace ROS2
         ConnectToLidarRaycaster();
         ConfigureLidarRaycaster();
 
-        m_sourceEventHandler = decltype(m_sourceEventHandler)([this](auto... args)
-        {
-            this->Visualize();
-        });
-        m_eventSourceAdapter.ConnectToSourceEvent(m_sourceEventHandler);
-
-        m_adaptedEventHandler = decltype(m_adaptedEventHandler)([this](auto... args)
-        {
-            this->FrequencyTick();
-        });
-        m_eventSourceAdapter.ConnectToAdaptedEvent(m_adaptedEventHandler);
-
-        m_eventSourceAdapter.Configure(m_sensorConfiguration);
-        m_eventSourceAdapter.Activate();
+        StartSensor(
+            m_sensorConfiguration.m_frequency,
+            [this]([[maybe_unused]] auto&&... args)
+            {
+                if (!m_sensorConfiguration.m_publishingEnabled)
+                {
+                    return;
+                }
+                FrequencyTick();
+            },
+            [this]([[maybe_unused]] auto&&... args)
+            {
+                Visualize();
```

Comment:
Lets discuss: 
"We should only visualise data with given frequency, otherwise the visualisation data will be misleading (even if looking smoother)"

Reply:
This will probably lead to points being visualized once every few frames - they will be not visible most of the time. If this is acceptable, I can change the code that way.